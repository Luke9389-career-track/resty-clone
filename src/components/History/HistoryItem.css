.HistoryItem{
  width: 100%;
  margin: 0.3em;
  padding: 0.3em;
  background: white;
  
}
