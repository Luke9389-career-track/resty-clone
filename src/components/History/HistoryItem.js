import React from 'react';
import PropTypes from 'prop-types';
import styles from './HistoryItem.css';

const HistoryItem = ({ handleClick, history }) => {
  return (
    <div
      className={styles.HistoryItem} 
      onClick={() => handleClick(history.url, history.method, history.headers, history.response)}>
      <h2>{`${history.method.toUpperCase()}`}</h2>
      <p>{`${history.url}`}</p>
    </div>
  );
};

HistoryItem.propTypes = {
  handleClick: PropTypes.func.isRequired,
  history: PropTypes.shape({
    url: PropTypes.string.isRequired,
    method: PropTypes.string.isRequired,
    headers: PropTypes.object.isRequired,
    response: PropTypes.object.isRequired,
  })
};


export default HistoryItem;
