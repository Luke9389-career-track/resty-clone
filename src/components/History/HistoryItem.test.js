import React from 'react';
import { shallow } from 'enzyme';
import HistoryItem from './HistoryItem';

describe('HistoryItem component', () => {
  const props = {
    handleClick: () => { },
    history: {
      url: 'ghj',
      method: 'hjk',
      headers: {},
      response: {}
    }
  };
  it('renders HistoryItem', () => {
    const wrapper = shallow(<HistoryItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
