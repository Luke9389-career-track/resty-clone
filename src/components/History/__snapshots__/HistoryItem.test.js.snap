// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryItem component renders HistoryItem 1`] = `
<div
  className="HistoryItem"
  onClick={[Function]}
>
  <h2>
    HJK
  </h2>
  <p>
    ghj
  </p>
</div>
`;
